integration/networking/bridge_test.go

```diff
@@ -0,0 +1,331 @@
+// assigned by the kernel work properly. To do so, the MAC address of ctr1 is manually assigned.
+func TestBridgeIPv6SLAACLLAddress(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+
+	d := daemon.New(t)
+	d.StartWithBusybox(t, "-D", "--experimental", "--ip6tables")
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	testcases := []struct {
+		name       string
+		bridgeOpts []func(create *types.NetworkCreate)
+	}{
+		{
+			name:       "IPv6 non-internal network",
+			bridgeOpts: []func(create *types.NetworkCreate){},
+		},
+		{
+			name: "IPv6 internal network",
+			bridgeOpts: []func(create *types.NetworkCreate){
+				network.WithInternal(),
+			},
+		},
+	}
+
+	for tcID, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx := context.Background()
+
+			bridgeName := fmt.Sprintf("testnet-ip6ll-%d", tcID)
+			network.CreateNoError(ctx, t, c, bridgeName, append(tc.bridgeOpts,
+				network.WithDriver("bridge"),
+				network.WithIPv6(),
+				network.WithIPAM("fdf1:a844:380c:b247::/64", "fdf1:a844:380c:b247::1"),
+				network.WithOption("com.docker.network.bridge.name", bridgeName))...)
+			defer network.RemoveNoError(ctx, t, c, bridgeName)
+
+			ctr1Name := sanitizeCtrName(t.Name() + "-ctr1")
+			id1 := container.Run(ctx, t, c,
+				container.WithName(ctr1Name),
+				container.WithImage("busybox:latest"),
+				container.WithCmd("/bin/sleep", "infinity"),
+				// Link-local address is derived from the MAC address, so we need to
+				// specify one here to hardcode the SLAAC LL address pinged below.
+				container.WithMacAddress("02:42:ac:11:00:02"),
+				container.WithNetworkMode(bridgeName))
+			defer c.ContainerRemove(ctx, id1, types.ContainerRemoveOptions{
+				Force: true,
+			})
+
+			ctr2Name := sanitizeCtrName(t.Name() + "-ctr2")
+			attachCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+			res := container.RunAttach(attachCtx, t, c,
+				container.WithName(ctr2Name),
+				container.WithImage("busybox:latest"),
+				container.WithCmd("ping", "-c1", "-W3", "-6", "fe80::42:acff:fe11:2%eth0"),
+				container.WithNetworkMode(bridgeName))
+			defer c.ContainerRemove(ctx, res.ContainerID, types.ContainerRemoveOptions{
+				Force: true,
+			})
+
+			assert.Check(t, res.ExitCode == 0)
+			assert.Check(t, res.Stderr.Len() == 0)
+			assert.Check(t, strings.Contains(res.Stdout.String(), "1 packets transmitted, 1 packets received"))
+
+			t.Logf("Logs from %s:\n%s", ctr2Name, res.Stdout.String())
+		})
+	}
+}
+
+// TestBridgeINC makes sure two containers on two different bridge networks can't communicate with each other.
+func TestBridgeINC(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+
+	d := daemon.New(t)
+	d.StartWithBusybox(t, "-D", "--experimental", "--ip6tables")
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	const bridge1Opts = "bridge1"
+	const bridge2Opts = "bridge2"
```

Comment:
Nit: define a struct type instead of using maps with a fixed set of keys. Types can be declared inside a function body, BTW.

Reply:
> Types can be declared inside a function body, BTW.

Yeah, I was hesitant on this one; didn't know what approach would be more idiomatic. Will fix that.